### ISSUE TITLE
The voice search is wrongly performed after closing the first search and quickly clicking the microphone button

### ISSUE THREAD

**Issue by Softvision-RemusDranca** [MozillaReality/FirefoxReality#2723]
**[Affected Versions]:**
- Firefox Reality version 9-40290101 (854287c0)

**[Affected Platforms]:**
- Oculus Quest - Android 7.1.1
- HTC Vive Focus Plus - Android 7.1.2

**[Prerequisites]:**
- The Firefox Reality browser is opened.
- The Microphone permissions are enabled.

**[Steps to reproduce]:**
1. Click the Microphone button.
2. Say a word and click the close button.
3. Quickly click the Microphone button again.
4. Observe the behavior

**[Expected results]:**
- No searches are performed until saying another word.

**[Actual results]:**
- The previous search is performed and the Voice Search prompt is dismissed.

**[Notes]:**
- This issue is also reproducible if the Voice Search prompt is dismissed while the search is in progress. After quickly clicking the Microphone button, the voice Search prompt is briefly displayed.
- Attached is a screen recording of the issue: [link](https://drive.google.com/open?id=1xCsfGSqPBl-cf-5dvOv4RnASAtIpaDAX).

**Comment by keianhzo** [comment]
Blocked by https://github.com/mozilla/androidspeech/pull/26

**Comment by keianhzo** [comment]
Still blocked, waiting for review. Pinging @andrenatal

**Comment by Softvision-GeluHaiduc** [comment]
I have verified this issue with the latest Firefox Reality version 9 RC4 (c64365e3) installed on the Oculus Go, Quest, HTC Vive Focus, Focus Plus and Pico Neo2 headsets.
Now, if I quickly click the voice search button again the voice search gets stuck and no voice search can be performed until the voice search is dismissed and reopened. [Here](https://drive.google.com/open?id=1-Re1fg_zdzeXFKYeh_jKI4BP7UBbAMUG) is a video of the behaviour.
Because of this new behaviour I will reopen the issue.

**Comment by keianhzo** [comment]
The way the speech library works right now makes it hard to handle this cases where a voice search is performed right after a cancel. The previous request is still ongoing after we start a new one so the latest start gets cancelled and we still get event from the previous request. This will need updating the  MozillaSpeechLibrary.

Given the timing, I think we should push this to v10 so we have time to work on a better way of handling the state inside MSL.

**Comment by keianhzo** [comment]
Waiting for https://github.com/mozilla/androidspeech/pull/34 to land

**Comment by keianhzo** [comment]
Still waiting for #3340

**Comment by Softvision-RemusDranca** [comment]
I have verified this issue and it's no longer reproducible using Firefox Reality version 11-141621406 (c7dd36c). Each session of voice search is now correctly performed and previous searches are not affecting the new ones.
However, a new issue occurs when trying to deny voice permissions. This is tracked in #3491.

Tested on HTC Vive Focus (Android 7.1.2), Oculus Quest (Android 7.1.1), and Pico Neo 2 (Android - 8.1.0) devices.
